crates/core/tedge_actors/src/servers/message_boxes.rs

```diff
@@ -80,69 +73,92 @@ impl<Request: Message, Response: Message> ConcurrentServerMessageBox<Request, Re
             return ControlFlow::Continue(());
         }
 
-        select! {
+        tokio::select! {
             Some(result) = self.pending_responses.next() => {
-                self.send_result(result).await;
+                if let Err(err) = result {
+                    log::error!("Request failed with: {err}");
+                }
                 ControlFlow::Continue(())
             },
             // recv consumes the message from the channel, so we can't just use
             // a regular return, because then next_request wouldn't see it
             //
             // a better approach would be to do select on top-level entry point,
             // then we'd be sure we're able to cancel when anything happens, not
-            // just when waiting for pending_responses, e.g. if send_result
-            // stalls
-            Some(RuntimeRequest::Shutdown) = self.clients.recv_signal() => {
+            // just when waiting for pending_responses.
+            Some(RuntimeRequest::Shutdown) = self.requests.recv_signal() => {
                 ControlFlow::Break(())
             }
             else => ControlFlow::Break(())
         }
     }
 
-    pub fn send_response_once_done(&mut self, pending_result: PendingResult<(ClientId, Response)>) {
+    pub fn send_response_once_done(&mut self, pending_result: PendingResult) {
```

Comment:
```suggestion
    pub fn register_request_handler(&mut self, pending_result: PendingResult) {
```
That's a misleading name, now that we're not sending the response here.